bin/hdfcoinc/pycbc_multiifo_combine_statmap

```diff
@@ -0,0 +1,182 @@
+    f.attrs['background_time_exc/{}'.format(ifosstring)] = fi.attrs['background_time_exc']
+    f.attrs['foreground_time_exc/{}'.format(ifosstring)] = fi.attrs['foreground_time_exc']
+
+    
+
+key_set = pycbc.io.name_all_datasets(files)
+# copy and concatenate all the columns in the foreground group 
+# from all files /except/ the IFO groups, we recalculate ifar/fap later
+for key in key_set:
+    if key.startswith('foreground') and not any([ifo in key for ifo in args.ifos]):
+        pycbc.io.combine_and_copy(f, files, key)
+
+
+logging.info('collating triggers into single structure')
+#for the ifo-specific groups, copy them over into the new file
+# - if the ifo is not present for each coincident trigger, 
+#  indicate by including a '-1' value in the combined file
+
+trigger_list = []
+all_trig_times = {}
+all_trig_ids = {}
+for ifo in args.ifos:
+    all_trig_times[ifo] = numpy.array([], dtype=numpy.uint32)
+    all_trig_ids[ifo] = numpy.array([], dtype=numpy.uint32)
+
+# for each file, append the trigger time and id data for each ifo
+#   - if it isnt there, then fill with -1 values
+for f_in in files:
+    for ifo in args.ifos:
+        if ifo in f_in['foreground']:
+            all_trig_times[ifo] = numpy.concatenate([all_trig_times[ifo], \
+                                     f_in['foreground/{}/time'.format(ifo)][:]])
+            all_trig_ids[ifo] = numpy.concatenate([all_trig_ids[ifo], 
+                                     f_in['foreground/{}/trigger_id'.format(ifo)][:]])
+        else:
+            all_trig_times[ifo] = numpy.concatenate([all_trig_times[ifo],
+                                     -1*numpy.ones_like(f_in['foreground/fap'][:], dtype=numpy.uint32)])
+            all_trig_ids[ifo] = numpy.concatenate([all_trig_ids[ifo],
+                                     -1*numpy.ones_like(f_in['foreground/fap'][:], dtype=numpy.uint32)])
+
+logging.info('clustering triggers for loudest stat value')
+#cluster the triggers before applying trials factor
+for ifo in args.ifos:
+    f['foreground/{}/time'.format(ifo)] = all_trig_times[ifo]
+    f['foreground/{}/trigger_id'.format(ifo)] = all_trig_ids[ifo]
+
+#cluster for loudest stat value 
+def argmax(v):
+    return numpy.argsort(v)[-1]
+
+stat = numpy.core.records.fromarrays([f['foreground/ifar'][:],
+                                      f['foreground/stat'][:]],
+                                      names='ifar,stat')
+
+all_times = ()
+for ifo in args.ifos:
+    all_times = all_times +  (f['foreground/%s/time' % ifo][:],)
+
+cidx = pycbc.events.cluster_coincs_multiifo(stat, all_times,
+                                            numpy.zeros(len(stat)), 0,
+                                            args.cluster_window, argmax)
+
+# downsample the foreground columns to only the loudest ifar between the multiple files
+for key in f['foreground'].keys():
+    if key not in args.ifos:
+        dset = f['foreground/%s' % key][:][cidx]
```

Comment:
No - because the f['foreground/%s' % key][:][cidx] is a different size to f['foreground/%s' % key][:], it has to be deleted and re-set

It seems like a silly workaround but I spend some time a few days ago trying to sort this to no avail